src/query/plan/cost_estimator.hpp

```diff
@@ -212,6 +214,31 @@ class CostEstimator : public HierarchicalLogicalOperatorVisitor {
     return true;
   }
 
+  bool PreVisit(Union &op) override {
+    double left_cost = EstimateCostOnBranch(&op.left_op_);
+    double right_cost = EstimateCostOnBranch(&op.right_op_);
+
+    // the number of hits in the previous operator should be the joined number of results of both parts of the union
+    cardinality_ *= (left_cost + right_cost);
```

Comment:
```suggestion
    // the number of hits in the previous operator should be the joined number of results of both parts of the union
    cardinality_ *= (EstimateCostOnBranch(&op.left_op_); + EstimateCostOnBranch(&op.right_op_););
```
Again, feel free to ignore it but I think it is not necessary to introduce local variables

Reply:
it's easier for debugging though as you get the returned value and you can set a breakpoint

Reply:
the prior suggestions you have suggested to commit make sense though